src/base/ceramicApiKit.ts

```diff
@@ -0,0 +1,167 @@
+
+interface ConfirmationNode {
+  owner: string;
+  id: string;
+  signature: string;
+  signatureType: string;
+  submissionDate: string;
+  transactionHash: string;
+  confirmationType: string;
+}
+
+interface ConfirmationEdge {
+  node: ConfirmationNode;
+}
+
+interface ConfirmationIndexData {
+  data: {
+    confirmationIndex: {
+      edges: ConfirmationEdge[];
+    };
+  }
+}
+
+export class ceramicApiKit {
+  private composeClient: any;
+  private privateKey: string;
+  private ceramicUrl: string;
+
+  constructor(privateKey: string, ceramicUrl: string) {
+    this.privateKey = privateKey;
+    this.ceramicUrl = ceramicUrl;
+  }
+
+  async connect() {
+    try {
+      const { ComposeClient } = await import('@composedb/client');
+      const composeClient = new ComposeClient({
+        ceramic: this.ceramicUrl,
+        //cannot import type from ESM only module in nest.js
+        // @ts-ignore
+        definition: definition,
+      });
+      const { fromString } = await import('uint8arrays/from-string');
+      const seedArray = fromString(this.privateKey, "base16");
+      const { Ed25519Provider } = await import('key-did-provider-ed25519');
+      const provider = new Ed25519Provider(new Uint8Array(seedArray));
+      const { DID } = await import('dids');
+      const { getResolver } = await import('key-did-resolver');
+      const did = new DID({ provider, resolver: getResolver() });
+      await did.authenticate();
+      composeClient.setDID(did);
+      this.composeClient = composeClient;
+    } catch (error) {
+      console.error('Error on connecting to Ceramic:', error);
+      throw error;
+    }
+  }
+
+  async proposeTransaction({
+                             safeAddress,
+                             safeTransactionData,
+                             safeTxHash,
+                             senderAddress,
+                             senderSignature,
+                             origin
+                           }: ProposeTransactionProps, threshold: number): Promise<void> {
+    if (!this.composeClient) {
+      await this.connect();
+    }
+    const confirmation = await this.composeClient.executeQuery(`
+            mutation CreateConfirmation {
+                createConfirmation(
+                    input: {
+                        content: {
+                            owner: "${senderAddress}"
+                            signature: "${senderSignature}"
+                            signatureType: "ECDSA"
+                            submissionDate: "${(new Date()).toISOString()}"
+                            transactionHash: "${safeTxHash}"
+                            confirmationType: "approve"
+                        }
+                        clientMutationId: null
+                    }
+                ) {
+                    clientMutationId
+                }
+            }
+        `);
+
+    const transaction = await this.composeClient.executeQuery(`mutation CreateTransaction {
```

Comment:
Please delete this storage if it's not used.

Reply:
thank you! the comment answers this question
https://github.com/helix-bridge/relayer/pull/30#discussion_r1682020468

Reply:
done @xiaoch05